mabwiser/base_mab.py

```diff
@@ -203,3 +247,143 @@ def _effective_jobs(size: int, n_jobs: int):
             n_jobs = max(mp.cpu_count() + 1 + n_jobs, 1)
         n_jobs = min(n_jobs, size)
         return n_jobs
+
+    @staticmethod
+    def _get_arm_distances(from_arm: Arm, arm_to_features: Dict[Arm, List[Num]], metric: str = 'cosine',
+                           self_distance: int = 999999) -> Dict[Arm, Num]:
+        """
+        Calculates the distances of the given from_arm to all the arms.
+
+        Distances calculated based on the feature vectors given in arm_to_features using the given distance metric.
+        The distance of the arm to itself is set as the given self_distance.
+
+        Parameters
+        ---------
+        from_arm: Arm
+            Distances from this arm.
+        arm_to_features: Dict[Arm, list[Num]]
+            Features for each arm used to calculate distances.
+        metric: str
+            Distance metric to use.
+            Default value is 'cosine'.
+        self_distance: int
+            The value to set as the distance to itself.
+            Default value is 999999.
+
+        Returns
+        -------
+        Returns distance from given arm to arm v as arm_to_distance[v].
+        """
+
+        # Find the distance of given from_arm to all arms including self
+        arm_to_distance = {}
+        for to_arm in arm_to_features.keys():
+            if from_arm == to_arm:
+                arm_to_distance[to_arm] = self_distance
+            else:
+                arm_to_distance[to_arm] = cdist(np.asarray([arm_to_features[from_arm]]),
+                                                np.asarray([arm_to_features[to_arm]]),
+                                                metric=metric)[0][0]
+
+                # Cosine similarity can be nan when a feature vector is all-zeros
+                if np.isnan(arm_to_distance[to_arm]):
+                    arm_to_distance[to_arm] = self_distance
+
+        return arm_to_distance
+
+    def _get_pairwise_distances(self, arm_to_features: Dict[Arm, List[Num]], metric: str = 'cosine',
```

Comment:
It uses `_get_arm_distances` which is a method of the class?

Reply:
That's also static, so you could say `BaseMAB._get_arm_distances` instead of `self._get_arm_distances`, right? Not sure if IDEs would complain that `BaseMAB` isn't initialized though.

Reply:
Updated.